File: exercism/go/leap/leap.go
/*
	Package leap contains a function that determins
	if a given year is a leap year.
*/
package leap

// IsLeapYear calculates if a given year is a leap year
func IsLeapYear(year int) bool {
	isaFourthYear := year%4 == 0
	isaHundredthYear := year%100 == 0
	isaFourthHundredthYear := year%400 == 0

	return (isaFourthYear && !isaHundredthYear) || isaFourthHundredthYear
}

File: exercism/fsharp/leap/Leap.fs
module Leap

let leapYear (year: int): bool =
  let isAFourthYear = year % 4 = 0
  let isNotAHundrethYear = year % 100 <> 0
  let isAFourthHundrethYear = year % 400 = 0
  (isAFourthYear && isNotAHundrethYear) || isAFourthHundrethYear
